contrib/macdeploy/README.md

```diff
@@ -14,44 +14,45 @@ When complete, it will have produced `Bitcoin-Qt.dmg`.
 
 ## SDK Extraction
 
+### Step 1: Obtaining `Xcode.app`
+
 Our current macOS SDK (`macOSX10.14.sdk`) can be extracted from
 [Xcode_10.2.1.xip](https://download.developer.apple.com/Developer_Tools/Xcode_10.2.1/Xcode_10.2.1.xip).
 An Apple ID is needed to download this.
 
-`Xcode.app` is packaged in a `.xip` archive.
-This makes the SDK less-trivial to extract on non-macOS machines.
-One approach (tested on Debian Buster) is outlined below:
+After Xcode version 7.x, Apple started shipping the `Xcode.app` in a `.xip`
+archive. This makes the SDK less-trivial to extract on non-macOS machines. One
+approach (tested on Debian Buster) is outlined below:
 
 ```bash
+# Install/clone tools needed for extracting Xcode.app
+apt install cpio
+git clone https://github.com/theuni/apple-sdk-tools.git
 
-apt install clang cpio git liblzma-dev libxml2-dev libssl-dev make
-
-git clone https://github.com/tpoechtrager/xar
-pushd xar/xar
-./configure
-make
-make install
-popd
-
-git clone https://github.com/NiklasRosenstein/pbzx
-pushd pbzx
-clang -llzma -lxar pbzx.c -o pbzx -Wl,-rpath=/usr/local/lib
-popd
-
-xar -xf Xcode_10.2.1.xip -C .
+# Unpack Xcode_10.2.1.xip and place the resulting Xcode.app in your current
+# working directory
+python3 apple-sdk-tools/extract_xcode.py -f Xcode_10.2.1.xip | cpio -d -i
+```
 
-./pbzx/pbzx -n Content | cpio -i
+On macOS the process is more straightforward:
 
-find Xcode.app -type d -name MacOSX.sdk -exec sh -c 'tar --transform="s/MacOSX.sdk/MacOSX10.14.sdk/" -c -C$(dirname {}) MacOSX.sdk/ | gzip -9n > MacOSX10.14.sdk.tar.gz' \;
+```bash
+xip -x Xcode_10.2.1.xip
 ```
 
-on macOS the process is more straightforward:
+### Step 2: Generating `MacOSX10.14.sdk.tar.gz` from `Xcode.app`
+
+To generate `MacOSX10.14.sdk.tar.gz`, run the script
+[`gen_sdk_package.sh`](./gen_sdk_package.sh) with the XCODEDIR environment
+variable pointing to the `Xcode.app` extracted in the previous stage.
 
 ```bash
-xip -x Xcode_10.2.1.xip
-tar -s "/MacOSX.sdk/MacOSX10.14.sdk/" -C Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/ -czf MacOSX10.14.sdk.tar.gz MacOSX.sdk
+# Generate a MacOSX10.14.sdk.tar.gz from the supplied Xcode.app
+./contrib/macdeploy/gen_sdk_package.sh '/path/to/Xcode.app'
 ```
 
+### Historial macOS SDK Extraction Notes
```

Comment:
I'm not sure keeping instructions around for previous SDKs is that useful. Only for the release that is used for building. Older information can always be found in git.

Reply:
Fixed!